packages/core/src/lib/position.mts

```diff
@@ -0,0 +1,49 @@
+type Change<T> = {
+  previous: T;
+  current: T;
+};
+
+type PageInfo = {
+  active: Change<number>;
+  total: number;
+  pageSize: number;
+};
+
+/**
+ * Given information about a page, decides the next position at which the active
+ * item should be rendered in the page.
+ */
+type PositionReducer = (info: PageInfo) => (pointer: number) => number;
```

Comment:
If `set` from `useState` someday (I hope soon) takes a function to do atomic updates instead of values, this will be even more useful